test/e2e/tests/test_scenarios.py

```diff
@@ -37,6 +37,88 @@ class ParameterGroupHelper(helper.ResourceHelper):
     Overrides methods as required for custom resources.
     """
 
+    def assert_items_k8s(self, expectations: dict, state: dict) -> None:
+        if not expectations:
+            # nothing to assert as there are no expectations
+            return
+        if not state:
+            # there are expectations but no given state to validate
+            # following assert will fail and assert introspection will provide useful information for debugging
+            assert expectations == state
+
+        for (key, value) in expectations.items():
+            # conditions are processed separately
+            if key == "conditions":
+                continue
+            # tags are processed by custom logic
+            if key == "tags":
+                for expectedTag in value:
+                    tagKeyExist = False
+                    tagKey = expectedTag.get("key")
+                    latestTags = state.get(key)
+                    for latestTag in latestTags:
+                        if ("key", tagKey) in latestTag.items():
+                            assert expectedTag == latestTag
+                            tagKeyExist = True
+                            break
+                    if tagKeyExist:
+                        continue
+                    assert expectedTag == None
+                continue
+            # parameterNameValues are processed by custom logic
+            if key == "parameterNameValues":
+                if not value:
+                    assert value == state.get(key)
+                    continue
+                for expectedParameterNameValue in value:
+                    parameterNameExist = False
+                    parameterName = expectedParameterNameValue.get("parameterName")
+                    latestParameterNameValues = state.get(key)
+                    for latestParameterNameValue in latestParameterNameValues:
+                        if ("parameterName", parameterName) in latestParameterNameValue.items():
+                            assert expectedParameterNameValue == latestParameterNameValue
+                            parameterNameExist = True
+                            break
+                    if parameterNameExist:
+                        continue
+                    assert expectedParameterNameValue == None
+                continue
+            assert value == state.get(key)
+
+    def assert_items_boto3(self, expectations: dict, state: dict) -> None:
+        if not expectations:
+            # nothing to assert as there are no expectations
+            return
+        if not state:
+            # there are expectations but no given state to validate
+            # following assert will fail and assert introspection will provide useful information for debugging
+            assert expectations == state
+
+        mdb = boto3.client("memorydb")
```

Comment:
You can put this line in the constructor of `ResourceHelper` and store it as a member variable, that way we aren't initialising it for every `assert` call.

Reply:
Modified.

Reply:
I've tried in python and function can pass parameters. However, when I tried to put `mdb` into `ResourceHelper`, it reports an error in other-resourceHelp(ResourceHelper) that `mdb is not defined`. Hence, I will keep this way.

Reply:
I am not satisfied with that answer. I am not talking about functions and parameters, I am talking about initialising a member variable inside the constructor of the parent class.

Reply:
Incorrectly set it to an instant variable. Put it into constructor now.